netbox/client.go

```diff
@@ -95,7 +120,38 @@ func (c *Client) NewRequest(method string, endpoint string, options Valuer) (*ht
 	}
 	u.RawQuery = values.Encode()
 
-	return http.NewRequest(method, u.String(), nil)
+	return http.NewRequest(method, u.String(), body)
+}
+
+// NewJSONRequest creates a HTTP request using the input HTTP method, URL
+// endpoint, a Valuer which creates URL parameters for the request, and
+// a io.Reader as the body of the request. For body, expecting some
+// json.Marshal-able struct. nil body is not allowed.
+// NewJSONRequest also sets HTTP Header
+// "Content-Type: application/json; utf-8"
+//
+// If a nil Valuer is specified, no query parameters will be sent with the
+// request.
+func (c *Client) NewJSONRequest(method string, endpoint string, options Valuer, body interface{}) (*http.Request, error) {
```

Comment:
This function body is a little dense.  I like to break up related blocks with whitespace where it makes sense, to improve readability.

How about logically separating into:
- parameter checking
- JSON marshaling
- request creation
- request finalization and return

Like:

```go
	if body == nil {
		return nil, errors.New("expected body to be not nil")
	}

	b := new(bytes.Buffer)
	err := json.NewEncoder(b).Encode(body)
	if err != nil {
		return nil, err
	}

	req, err := c.NewDataRequest(
		method,
		endpoint,
		options,
		b,
	)
	if err != nil {
		return nil, err
	}

	req.Header.Set("Content-Type", "application/json; charset=utf-8")
	return req, nil
```